lib/awful/placement.lua

```diff
@@ -702,23 +704,35 @@ local function get_relative_regions(geo, mode, is_absolute)
         end
     end
 
-    -- Get the drawable geometry
-    local dpos = geo.drawable and (
-        geo.drawable.drawable and
-            geo.drawable.drawable:geometry()
-            or geo.drawable:geometry()
-    ) or {x=0, y=0}
+    -- Get the parent geometry using one way or another depending on the object
+    -- Type
+    local bw, dgeo = 0, {x=0, y=0, width=1, height=1}
```

Comment:
why initial width and height are set to 1?

Reply:
Because height and width or 0 are not supported by objects such as clients. For consistency, the default geometry is `{0,0,1,1}`.